### PR TITLE
lxddoc: Within a config group, alphabetically sort the config options

### PR DESCRIPTION
**Pull request by gabrielmougard**
Alphabetically sort the config options within a config group

### CODE REVIEW

File: lxd/config/generate/lxddoc.go [hunk 1]
```diff
@@ -191,7 +192,38 @@ func parse(path string, outputYAMLPath string, excludedPaths []string) (*doc, er
...
 	}
 
-	yamlDoc.Configs = projectEntries
+	// Within each group, alphabetically sort the entries by key (config option key)
+	orderedProjectEntries := make(map[string][]any, len(projectEntries))
+	for groupKey, entries := range projectEntries {
+		var sortedConfigOptionKeysPerGroup []string
+		for _, configEntry := range entries {
+			c := configEntry.(map[string]any)
+			for k := range c {
```
Comment (monstermunchkin) [inline lxd/config/generate/lxddoc.go:201]:
```suggestion
			for k := range configEntry.(map[string]any) {
```
There's no need for the `c` variable.

File: lxd/config/generate/lxddoc.go [hunk 2]
```diff
@@ -191,7 +192,38 @@ func parse(path string, outputYAMLPath string, excludedPaths []string) (*doc, er
...
+				sortedConfigOptionKeysPerGroup = append(sortedConfigOptionKeysPerGroup, k)
+			}
+		}
+
+		sort.Strings(sortedConfigOptionKeysPerGroup)
+		for _, configOptionKey := range sortedConfigOptionKeysPerGroup {
+			for _, configEntry := range entries {
+				c := configEntry.(map[string]any)
+				for k := range c {
```
Comment (monstermunchkin) [inline lxd/config/generate/lxddoc.go:210]:
```suggestion
				for k := range configEntry.(map[string]any) {
```

**Review by monstermunchkin — CHANGES_REQUESTED** [review]
Could this be put into a separate function? That way, we could add unit tests to ensure the function does what it's supposed to do.

File: lxd/config/generate/lxddoc.go (hunk 2 above)
Comment (gabrielmougard) [reply lxd/config/generate/lxddoc.go]: Actually, `c` is further used within the loop

**Comment by gabrielmougard** [comment]
@monstermunchkin added a unit test for the sorting function

File: lxd/config/generate/lxddoc.go (hunk 2 above)
Comment (monstermunchkin) [reply lxd/config/generate/lxddoc.go]: Oh right, I overlooked that.

**Review by monstermunchkin — APPROVED** [review]

**Comment by gabrielmougard** [comment]
thanks

**Review by ru-fu — APPROVED** [review]
Works as expected, thanks! :)

This is ready for merging @tomponline

**Pull request merged**